bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieServer.java

```diff
@@ -161,11 +161,11 @@ public void resumeProcessing() {
     }
 
     public synchronized void shutdown() {
+        LOG.info("Shutting down BookieServer");
```

Comment:
in BookieServer.start method, 'running' is set to true after nettyserver is started, so in BookieServer.shutdown method, nettyserver shutdown should be called atleast before the 'running' check. here i want to make sure that port is freed when BookieServer shutdown is called by not relying on 'running' status, which according to me is not reliable in all edge cases while testing. My main concern with 'running' variable is it is not reliable in the case of partial successful start (partial failure).

moreover I think shutdown methods should be idempotent operation and all the resources shutdown should be called irrespective of 'running' status. Ideally in shutdown methods, all they have to check is if the resource is null or not and if it is not null then it should call shutdown of that resource. Let that resource shutdown logic take care of all the scenarios, like if the resource is not started at all or if it is partially started.